app/router/group.py

```diff
@@ -29,10 +29,18 @@ def get_group_data(group_id: str):
         "headers": null
     }
     """
-    query_params = {"name": group_id, "type": "group"}
+    query_params = {}
+    for key in request.query_params.keys():
+        if key not in ["name", "id"]:
+            raise HTTPException(
+                status_code=400, detail="Query Parameter {} is not allowed!".format(key)
+            )
+        query_params[key] = request.query_params[key]
+    query_params["type"] = "group"
+
     response = requests.get(group_db_url, params=query_params)
     if response.status_code == 200:
         if len(response.json()) != 0:
-            return response.json()
-        return HTTPException(status_code=404, detail="Group does not exist!")
-    return HTTPException(status_code=response.status_code, detail=response.errors)
+            return response.json()[0]
+        raise HTTPException(status_code=404, detail="Group does not exist!")
```

Comment:
Why `raise` vs `return`?

Reply:
The docs recommend raising. https://fastapi.tiangolo.com/tutorial/handling-errors/